src/frequenz/sdk/config/_config_managing.py

```diff
@@ -150,51 +189,45 @@ async def _run(self) -> None:
         """
         await self.send_config()
 
-        parent_paths = {p.parent for p in self._config_paths}
-
-        # FileWatcher can't watch for non-existing files, so we need to watch for the
-        # parent directories instead just in case a configuration file doesn't exist yet
-        # or it is deleted and recreated again.
-        file_watcher = FileWatcher(
-            paths=list(parent_paths),
-            event_types=self._event_types,
-            force_polling=self._force_polling,
-            polling_interval=self._polling_interval,
-        )
-
-        try:
-            async for event in file_watcher:
-                # Since we are watching the whole parent directories, we need to make
-                # sure we only react to events related to the configuration files we
-                # are interested in.
-                if not any(event.path.samefile(p) for p in self._config_paths):
-                    continue
-
-                match event.type:
-                    case EventType.CREATE:
-                        _logger.info(
-                            "%s: The configuration file %s was created, sending new config...",
-                            self,
-                            event.path,
-                        )
-                        await self.send_config()
-                    case EventType.MODIFY:
-                        _logger.info(
-                            "%s: The configuration file %s was modified, sending update...",
-                            self,
-                            event.path,
-                        )
-                        await self.send_config()
-                    case EventType.DELETE:
-                        _logger.info(
-                            "%s: The configuration file %s was deleted, ignoring...",
-                            self,
-                            event.path,
-                        )
-                    case _:
-                        assert_never(event.type)
-        finally:
-            del file_watcher
+        async for event in self._file_watcher:
+            # Since we are watching the whole parent directories, we need to make
+            # sure we only react to events related to the configuration files we
+            # are interested in.
+            #
+            # pathlib.Path.samefile raises error if any path doesn't exist so we need to
+            # make sure `event.path` and `p` exists before calling it.
+            #
+            # On EventType.DELETE event the event.path won't exists and we have no way
+            # to check if this path is in self._config_paths so we need to skip this
+            # event.
+            if not event.path.exists() or not any(
+                event.path.samefile(p) for p in self._config_paths if p.exists()
+            ):
+                continue
```

Comment:
This is also much clearer if we don't subscribe to `DELETE`, then this can only happen on `CREATE` or `MODIFY`. If the file not there when the event was triggered, it means there was a race, and the file was created or changed and right after that it was deleted, so we can more safely (or more clearly) ignore the event, and hope a subsequent `CREATE` will be triggered again when the file is re-created.

If any of `event.path` or `self._config_paths` do not exist, I would log an error though. It might be a race, but I think we should be notified a config file was removed. We can then see if we get a lot of spurious errors if we want to remove it, o maybe set a timer and if the file doesn't come back after some time to then log an error.

Reply:
I thouht it is ok if not all files in self._config_paths exist

Reply:
My current understanding is : at least one file in `self._config_paths` must exists.

Reply:
> I thought it is ok if not all files in self._config_paths exist.

Yeah, true. As things are currently working it would be valid if only one file is missing. So if at some point we require all, we can update this too.

Reply:
* https://github.com/frequenz-floss/frequenz-sdk-python/issues/1120